### PR TITLE
CORE-19378 Wrapping repo uses highest gen key

### PR DESCRIPTION
**Pull request by simon-johnson-r3**
There are only two functional changes here. `findKeyAndId` and `findKeysWrappedByParentKey` (renamed from `findKeysWrappedByAlias`) now return only the latest generation for each wrapping key per alias, plus tests for that. The other changes are for tidying/code clarity purposes.

### CODE REVIEW

File: components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/WrappingRepositoryImpl.kt [hunk 1]
```diff
@@ -52,26 +52,31 @@ class WrappingRepositoryImpl(
...
         }
 
-    override fun findKeysWrappedByAlias(alias: String): List<WrappingKeyInfo> =
+    override fun findKeysWrappedByParentKey(parentKeyAlias: String): List<WrappingKeyInfo> =
         entityManagerFactory.createEntityManager().use {
             it.createQuery(
-                "FROM ${WrappingKeyEntity::class.simpleName} AS k WHERE k.parentKeyReference = :alias",
+                "FROM ${WrappingKeyEntity::class.simpleName} AS k WHERE k.parentKeyReference = :parentKeyAlias",
                 WrappingKeyEntity::class.java
-            ).setParameter("alias", alias).resultList.map { dao -> dao.toDto() }
```
Comment (anien) [inline components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/WrappingRepositoryImpl.kt:67]: Did we have a bug here that we were setting parameter `"alias"` to `alias` and not setting parameter `parentKeyAlias`?

Comment (simon-johnson-r3) [reply components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/WrappingRepositoryImpl.kt:67]: No. If you look at old line 65 it always stated: `WHERE k.parentKeyReference = :alias` it was just the local var here that was confusingly named. Alias is used everywhere else in this file and the interface to mean wrapping key alias.

Comment (anien) [reply components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/WrappingRepositoryImpl.kt:67]: Thanks for explanation.

**Review by anien — DISMISSED** [review]
LGTM.

**Review by anien — DISMISSED** [review]
LGTM.

**Review by anien — APPROVED** [review]

**Pull request merged**